src/Eloquent/Mixins/BuilderReturning.php

```diff
@@ -0,0 +1,113 @@
+                return $this->updateReturning([$deletedAtColumn => $time], $returning);
+            }
+
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->deleteReturning(null, $returning)->all()
+            )->each(fn (Model $model) => $model->exists = false);
+        };
+    }
+
+    public function insertOrIgnoreReturning(): Closure
+    {
+        return function (array $values, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->insertOrIgnoreReturning($values, $returning)->all()
+            );
+        };
+    }
+
+    public function insertReturning(): Closure
+    {
+        return function (array $values, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->insertReturning($values, $returning)->all()
+            );
+        };
+    }
+
+    public function insertUsingReturning(): Closure
+    {
+        return function (array $columns, $query, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->insertUsingReturning($columns, $query, $returning)->all()
+            );
+        };
+    }
+
+    public function updateFromReturning(): Closure
+    {
+        return function (array $values, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->updateFromReturning($values, $returning)->all()
+            );
+        };
+    }
+
+    public function updateOrInsertReturning(): Closure
+    {
+        return function (array $attributes, array $values = [], array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->updateOrInsertReturning($attributes, $values, $returning)->all()
+            );
+        };
+    }
+
+    public function updateReturning(): Closure
+    {
+        return function (array $values, array $returning = ['*']): Collection {
+            /** @var \Illuminate\Database\Eloquent\Builder $this */
+            $time = $this->getModel()->freshTimestampString();
+            $modelHelper = app(ModelHelper::class);
+
+            if ($updatedAtColumn = $modelHelper->getUpdatedAtColumn($this->getModel())) {
+                $values = array_merge([$updatedAtColumn => $time], $values);
+            }
+
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->updateReturning($values, $returning)->all()
+            );
```

Comment:
We don't need to do it by ourselves : 
```suggestion
            return $this->hydrate(
                $this->applyScopes()->getQuery()->updateReturning($this->addUpdatedAtColumn($values), $returning)->all()
            );
```